web3/contract/base_contract.py

```diff
@@ -865,43 +888,45 @@ def _prepare_transaction(
     @classmethod
     def _find_matching_fn_abi(
         cls,
-        fn_identifier: Optional[str] = None,
-        args: Optional[Any] = None,
-        kwargs: Optional[Any] = None,
-    ) -> ABIFunction:
-        return find_matching_fn_abi(
-            cls.abi, cls.w3.codec, fn_identifier=fn_identifier, args=args, kwargs=kwargs
+        fn_identifier: Optional[FunctionIdentifier] = None,
+        *args: Optional[Sequence[Any]],
```

Comment:
The defaults cause an error: `Parameter with "*" or "**" cannot have default value`. I haven't found an example where this would be possible as an `Optional`, so these `Optional` types don't make sense here.

I think the function itself will always initialize these with values internally, so it probably means we should just get rid of the `Optional`.